front/lib/swr/data_source_views.ts

```diff
@@ -143,18 +145,27 @@ export function useDataSourceViewContentNodes({
     }); // Serialize with body to ensure uniqueness.
   }, [url, body]);
 
-  const { data, error, mutate } = useSWRWithDefaults(fetchKey, async () => {
-    if (!url) {
-      return undefined;
-    }
-
-    return postFetcher([url, { internalIds, parentId, viewType }]);
-  });
+  const { data, error, mutate, mutateRegardlessOfQueryParams } =
+    useSWRWithDefaults(
+      fetchKey,
+      async () => {
+        if (!url) {
+          return undefined;
+        }
+
+        return postFetcher([url, { internalIds, parentId, viewType }]);
+      },
+      {
+        disabled: disabled || !viewType,
```

Comment:
does it make sense to let viewType be an optional arg ?

Reply:
Yes, it's undefined at first

Reply:
So convenient to let it disable the fetching